server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -436,6 +440,15 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
                     throw new IllegalArgumentException("Cannot have more than one plugin implementing a REST wrapper");
                 }
                 restWrapper = newRestWrapper;
+                if (restWrapper.getClass().getCanonicalName() == null ||
+                    restWrapper.getClass().getCanonicalName().startsWith("org.elasticsearch") == false) {
+                    deprecationLogger.deprecate(DeprecationCategory.SECURITY, "3rd_party_rest_deprecation", "The " +
```

Comment:
I think `DeprecationCategory.PLUGINS` is more appropriate than `DeprecationCategory.SECURITY`, since as Tim pointed out, this isn't a security feature directly.